File: Day 1/4/findMaximum.js
const findMaximum = (arr) => {
    return Math.max(...arr);
}

const findMaximumAlgorithmically = (arr) => {
    let max = arr[0];
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
    }
    return max;
}

console.log(findMaximum([1, 2, 3, 4, 5]));
console.log(findMaximumAlgorithmically([1, 2, 3, 4, 5]));

File: Day 1/4/find_maximum.c
#include <stdio.h>

int findMax(int arr[], int len) {
    int max = arr[0];
    for (int i = 1; i < len; i++) {
        if (arr[i] > max) {
            max = arr[i];
        }
    }
    return max;
}

int main() {
    int len;

    printf("Enter the size of the array: ");
    scanf("%d", &len);

    int arr[len];
    printf("Enter %d integers:\n", len);
    for (int i = 0; i < len; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Maximum number: %d\n", findMax(arr, len));
    return 0;
}
